providers/autodns/autoDnsProvider.go

```diff
@@ -0,0 +1,254 @@
+package autodns
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/pkg/txtutil"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+)
+
+var features = providers.DocumentationNotes{
+	providers.DocCreateDomains:       providers.Cannot(),
+	providers.DocDualHost:            providers.Cannot(),
+	providers.DocOfficiallySupported: providers.Cannot(),
+	providers.CanGetZones:            providers.Can(),
+	providers.CanUseAlias:            providers.Can(),
+	providers.CanUseCAA:              providers.Cannot(),
+	providers.CanUseDS:               providers.Cannot(),
+	providers.CanUsePTR:              providers.Cannot(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseSSHFP:            providers.Cannot(),
+	providers.CanUseTLSA:             providers.Cannot(),
+}
+
+type autoDnsProvider struct {
+	baseURL          url.URL
+	defaultHeaders   http.Header
+}
+
+func init() {
+	fns := providers.DspFuncs{
+		Initializer:   New,
+		RecordAuditor: AuditRecords,
+	}
+	providers.RegisterDomainServiceProviderType("AUTODNS", fns, features)
+}
+
+// New creates a new API handle.
+func New(settings map[string]string, _ json.RawMessage) (providers.DNSServiceProvider, error) {
+	api := &autoDnsProvider{}
+
+	api.baseURL = url.URL{
+		Scheme:      "https",
+		User:        url.UserPassword(
+			settings["username"],
+			settings["password"],
+			),
+		Host:        "api.autodns.com",
+		Path:        "/v1/",
+	}
+
+	api.defaultHeaders = http.Header{
+		"Accept": []string{"application/json; charset=UTF-8"},
+		"Content-Type": []string{"application/json; charset=UTF-8"},
+		"X-Domainrobot-Context": []string{settings["context"]},
+	}
+
+	return api, nil
+}
+
+// AuditRecords returns an error if any records are not
+// supportable by this provider.
+func AuditRecords(records []*models.RecordConfig) error {
```

Comment:
Please put `AuditRecords()` in `providers/autodns/auditrecords.go`  (we use the same file name in all providers so that automated updates are easier)

Reply:
Fixed in https://github.com/StackExchange/dnscontrol/pull/1336/commits/0c197ca17924e4f1484a54724b5759819f0296eb